rct229/rulesets/ashrae9012019/ruleset_functions/get_spaces_served_by_swh_use.py

```diff
@@ -0,0 +1,37 @@
+from rct229.utils.jsonpath_utils import find_all
+
+
+def get_spaces_served_by_swh_use(rmd: dict, swh_use_id: str) -> list[str]:
+    """
+    This function determines the spaces served by a given SWH use. The convention is that if any spaces reference the swh_use, then the service water heating use applies to only those spaces.
+    If no spaces reference the service water heating use, it applies to all spaces in the building segment.
+
+    Parameters
+    ----------
+    rmd: dict
+        A dictionary representing a ruleset model description as defined by the ASHRAE229 schema
+    swh_use_id: str
+        id of the `service_water_heating_uses` key
+
+    Returns
+    -------
+    spaces_served: list of str
+        list of space ids that has the sane service_water_heating_uses value
+    """
+
+    spaces_served = [
+        space["id"]
+        for space in find_all(
+            "$.buildings[*].building_segments[*].zones[*].spaces[*]", rmd
+        )
+        for swh_use in find_all("$.service_water_heating_uses[*]", space)
+        if swh_use_id == swh_use["id"]
+    ]
```

Comment:
hmmm, I wonder what is the point of a list?
This is because every service water heating use data will be a nested data in a space...

Reply:
Propose to move water heating uses to building segment level so we preserve this logic to minimize future impact.